compiler/damlc/tests/src/DA/Test/Packaging.hs

```diff
@@ -504,11 +505,110 @@ lfVersionTests damlc = testGroup "LF version dependencies"
+          step "Build proja"
+          createDirectoryIfMissing True (proja </> "src")
+          writeFileUTF8 (proja </> "src" </> "A.daml") $ unlines
+              [" daml 1.2"
+              , "module A where"
+              , "import DA.Text"
+              , "data A = A Int deriving Show"
+              -- This ensures that we have a reference to daml-stdlib and therefore daml-prim.
+              , "x : [Text]"
+              , "x = lines \"abc\\ndef\""
+              ]
+          writeFileUTF8 (proja </> "daml.yaml") $ unlines
+              [ "sdk-version: " <> sdkVersion
+              , "name: proja"
+              , "version: 0.0.1"
+              , "source: src"
+              , "dependencies: [daml-prim, daml-stdlib]"
+              ]
+          withCurrentDirectory proja $ callProcessSilent damlc ["build", "--target=1.6", "-o", proja </> "proja.dar"]
+          projaPkgIds <- darPackageIds (proja </> "proja.dar")
+          -- daml-stdlib, daml-prim and proja
+          length projaPkgIds @?= numStable16Packages + 2 + 1
+
+          step "Build projb"
+          createDirectoryIfMissing True (projb </> "src")
+          writeFileUTF8 (projb </> "src" </> "B.daml") $ unlines
+              [ "daml 1.2"
+              , "module B where"
+              , "import A"
+              , "data B = B A"
+              ]
+          writeFileUTF8 (projb </> "daml.yaml") $ unlines
+              [ "sdk-version: " <> sdkVersion
+              , "name: projb"
+              , "version: 0.0.1"
+              , "source: src"
+              , "dependencies: [daml-prim, daml-stdlib]"
+              , "data-dependencies: [" <> show (proja </> "proja.dar") <> "]"
+              ]
+          -- TODO Users should not have to pass --hide-all-packages, see https://github.com/digital-asset/daml/issues/4094
+          withCurrentDirectory projb $ callProcessSilent damlc
+            [ "build", "--target=1.7", "-o", projb </> "projb.dar"
+            , "--hide-all-packages"
+            , "--package", "(\"daml-prim\", True, [])"
+            , "--package", "(\"" <> damlStdlib <> "\", True, [])"
+            , "--package", "(\"proja-0.0.1\", True, [])"
+            ]
+          callProcessSilent repl ["validate", projb </> "projb.dar"]
+          projbPkgIds <- darPackageIds (projb </> "projb.dar")
+          -- daml-prim, daml-stdlib for 1.6, daml-prim, daml-stdlib for 1.7, proja and projb
+          length projbPkgIds @?= numStable17Packages + 2 + 2 + 1 + 1
+          -- daml-prim, daml-stdlib for 1.7, 2 stable 1.7 packages and projb
+          length (filter (`notElem` projaPkgIds) projbPkgIds) @?= 4
+    ] <>
+    [ testCaseSteps "Source generation edge cases" $ \step -> withTempDir $ \tmpDir -> do
+      writeFileUTF8 (tmpDir </> "Foo.daml") $ unlines
+        [ "daml 1.2"
+        , "module Foo where"
+        , "template Bar"
+        , "   with"
+        , "     p : Party"
+        , "     t : (Text, Int)" -- check for correct tuple type generation
```

Comment:
In what sense are we checking correctness here?

Reply:
Dunno tbh, this test just got shuffled around not changed. I think it is supposed to check that `Tuple2` is mapped back to a DAML tuple.